src/variant.ts

```diff
@@ -383,6 +384,20 @@ export class VariantManager implements vscode.Disposable {
       if (!chosen) {
         return false;
       }
+
+      const cfg = vscode.workspace.getConfiguration('cmake', this.folder.uri).inspect<object>('defaultVariants');
+      const telemetryProperties: telemetry.Properties = {
+        customFile: (util.checkFileExists("cmake-variants.json") ||
```

Comment:
Instead of checking file paths each time, can this state be saved during `_reloadVariantsFile` and then reported here?

Reply:
An in any case, these functions return promises and would need to be awaited.

Reply:
And in any case, these functions return promises and would need to be awaited.

Reply:
You are right, there is no need to check for the variants files and the defaultVariants setting every time a new variant is selected. It can all be moved to _reloadVariantsFile (no state var to be saved), do you agree?

Reply:
My mistake, _reloadVariantsFile is called too many times. I understand, a state var can be saved then and only when changing a variant we can report the telemetry. Exactly what you suggested.